core/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/kubernetesModels.scala

```diff
@@ -0,0 +1,106 @@
+package org.broadinstitute.dsde.workbench.leonardo
+
+import enumeratum.{Enum, EnumEntry}
+import org.broadinstitute.dsde.workbench.google2.GKEModels.{KubernetesClusterId, KubernetesClusterName, NodepoolName}
+import org.broadinstitute.dsde.workbench.google2.KubernetesModels.KubernetesApiServerIp
+import org.broadinstitute.dsde.workbench.google2.KubernetesSerializableName.KubernetesNamespaceName
+import org.broadinstitute.dsde.workbench.google2.{Location, MachineTypeName, NetworkName, SubnetworkName}
+import org.broadinstitute.dsde.workbench.model.WorkbenchEmail
+import org.broadinstitute.dsde.workbench.model.google.GoogleProject
+
+/** Google Container Cluster statuses
+ *  see: https://cloud.google.com/kubernetes-engine/docs/reference/rest/v1/projects.locations.clusters#Cluster.Status
+ */
+
+case class KubernetesCluster(id: KubernetesClusterLeoId,
+                             googleProject: GoogleProject,
+                             clusterName: KubernetesClusterName,
+                             location: Location,
+                             status: KubernetesClusterStatus,
+                             serviceAccountInfo: WorkbenchEmail,
+                             samResourceId: KubernetesClusterSamResource,
+                             auditInfo: AuditInfo,
+                             asyncFields: KubernetesClusterAsyncFields,
+                             namespaces: Set[KubernetesNamespaceName],
+                             labels: LabelMap,
+                             nodepools: Set[Nodepool]
+                            )
+
+object KubernetesCluster {
+  implicit class EnrichedKubernetesCluster(cluster: KubernetesCluster) {
+    def getGKEClusterId: KubernetesClusterId = KubernetesClusterId(cluster.googleProject, cluster.location, cluster.clusterName)
```

Comment:
Naming-wise, it'd be nice to follow what we've done with `GCE`, which is not capitalizing `C` and `E` (i.e. `getGkeClusterId`)